sdk/metric/internal/instrument/descriptor.go

```diff
@@ -0,0 +1,24 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instrument // import "go.opentelemetry.io/otel/sdk/metric/internal/instrument"
+
+import "go.opentelemetry.io/otel/sdk/instrumentation"
+
+type Description struct {
+	Library instrumentation.Library
+
+	InstrumentName        string
```

Comment:
Is this field the name of a metric (output) as well as an instrument (input)?

Reply:
Yes, they are one and the same.  One instrument can at most create one Metric within one view, it can be represented in multiple views and create multiple Metrics.

The follow-up question is will we combine multiple Metrics from different views?  No, I don't plan on it, that only happens when there is a conflict and is already an error state that we should be warning the users about.  If the user requests the same instrument.Description (so same scope and instrument name description) then we should just return the same instrument.